### PR TITLE
feat(quantic): add support for custom facet value captions

### PR DESCRIPTION
**Pull request by lbergeron**
https://coveord.atlassian.net/browse/SFINT-4847

This PR adds support for custom facet value captions in Quantic. The feature is available for standard and category facets. The custom captions are also displayed in facet search, and in the breadcrumb manager.

The custom captions are configured by assigning caption provider components to the `captions` named slot. Quantic provides the Facet Caption component out of the box to do that, but custom components can also be used to provide a set of custom captions at a time.

In all cases, a caption provider component MUST expose a `captions` property that returns an object mapping the facet values to the custom captions like so:

```javascript
{
  text: 'Plain text',
  html: 'Web page'
}
```

Custom captions are affected to a facet this way:

```html
<c-quantic-facet field="filetype" ...>
  <c-quantic-facet-caption slot="captions" value="text" caption="Plain text"></c-quantic-facet-caption>
  <c-my-own-caption-provider slot="captions"></c-my-own-caption-provider>
</c-quantic-facet>
```

### CODE REVIEW

File: packages/quantic/cypress/e2e/facets-1/facet/facet-with-captions.cypress.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import {FacetActions as Actions} from './facet-actions';
+import {FacetExpectations as Expect} from './facet-expectations';
+
+interface FacetCaptionOptions {
+  useCase: string;
+  value: string;
+  caption: string;
+}
+
+describe('Facet with captions Test Suite', () => {
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/facets-1/facet/facet-with-captions.cypress.ts:19]: Just for consistency with category facet test suit:
```suggestion
describe('quantic-facet with captions', () => {
```

**Review by mmitiche — APPROVED** [review]
Very neat! well done 👌

**Review by mmitiche — COMMENTED** [review]
What do you think about giving our customers a real life example by implementing this new feature in one of the standard/category facets of the ExampleSearch component?

The File Type facet seems like a good candidate:
File Type Facet

File: packages/quantic/force-app/main/default/lwc/quanticFacetCaption/quanticFacetCaption.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {api, LightningElement} from 'lwc';
+
+export default class QuanticFacetCaption extends LightningElement {
```
[the remark this answers is not in the document]
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticFacetCaption/quanticFacetCaption.js:3]: indeed if it doesn't have JSdoc it won't be included our reference doc.

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 3]
```diff
@@ -32,6 +31,11 @@ import expandFacet from '@salesforce/label/c.quantic_ExpandFacet';
...
  * @property {number} [index]
  */
+/**
+ * @typedef CaptionProvider
+ * @type {object}
+ * @property {Record<string, string>} captions
+ */
 
 /**
  * A facet is a list of values for a certain field occurring in the results, ordered using a configurable criterion (e.g., number of occurrences).
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:41]: Same comment regarding the JS doc comment to  obtain automatic documentation of the slot `captions`.

File: packages/quantic/force-app/main/default/lwc/quanticFacetCaption/quanticFacetCaption.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lbergeron) [reply packages/quantic/force-app/main/default/lwc/quanticFacetCaption/quanticFacetCaption.js:3]: You are right, I totally forgot the JSDoc.

**Review by nathanlb — APPROVED** [review]
Great work!

**Pull request merged**